catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java

```diff
@@ -214,6 +199,24 @@ public State complete(DeliveryManifestHolder deliveryManifestHolder)
         return State.COMPLETED;
     }
 
+    /**
+     * Uses CORBA exception structure to determine type of destination stored in incoming
+     * Destination object. Uses polymorphism to return the more specific type used, and throws
+     * an exception if not the correct type.
+     */
+
+    protected DestinationSink DestinationTypeChecker(Destination dest) throws Exception {
+        try {
```

Comment:
There is a `destination.descriminator()` that can be used to determine what type of destination has been set.